pybamm/expression_tree/operations/evaluate_python.py

```diff
@@ -491,21 +491,17 @@ def __init__(self, symbol):
         compiled_function = compile(python_str, result_var, "exec")
         exec(compiled_function)
 
-    def evaluate(self, t=None, y=None, y_dot=None, inputs=None, known_evals=None):
+    def __call__(self, t=None, y=None, inputs=None):
```

Comment:
why no more `y_dot`?

Reply:
I removed it because I was focused on making this consistent with the casadi evaluator. I remember adding it in because I was planning for support for full dae equations rather than just semi-explicit form, but this never really went forward. Happy to add it back if we still do want to go in that direction?

Reply:
Ok, sticking to semi-explicit is fine by me